examples/multi_region/README.md

```diff
@@ -0,0 +1,101 @@
+# Integrate EKS cluster(s) Audit Logs with Lacework - Multi-region
+
+This example creates all the required resources, as well as an IAM Role with a cross-account policy to 
+provide Lacework read-only access to monitor the audit logs.
+
+Additionally, this example creates multiple aws providers with an alias to create Cloudwatch Subscription filters 
+across multiple regions.
+
+## Inputs
+
+| Name                        | Description                                                                                     | Type           |
+| --------------------------- | ----------------------------------------------------------------------------------------------- | -------------- |
+| `cloudwatch_regions`        | A list of regions, to allow Cloudwatch Logs to be streamed from                                 | `list(string)` |
+| `no_cw_subscription_filter` | Set to true to create an integration with no Cloudwatch Subscription filter for your cluster(s) | `bool`         |
+
+## Sample Code
+
+```terraform
+provider "lacework" {}
+
+provider "aws" {
+  region = "eu-west-1"
+  alias  = "eu-west-1"
+}
+
+provider "aws" {
+  region = "us-west-2"
+  alias  = "us-west-2"
+}
+
+module "aws_eks_audit_log" {
+  source  = "lacework/eks-audit-log"
+  version = "~> 1.0"
+  cloudwatch_regions        = ["eu-west-1", "us-west-2"]
+  no_cw_subscription_filter = true
+}
+
+resource "aws_cloudwatch_log_subscription_filter" "lacework_cw_subscription_filter-eu-west" {
+  for_each = toset(["cluster-1", "cluster-2"])
+  provider = aws.eu-west-1
+
+
+  name            = "${module.aws_eks_audit_log.filter_prefix}-${each.value}"
+  role_arn        = module.aws_eks_audit_log.cloudwatch_iam_role_arn
+  log_group_name  = "/aws/eks/${each.value}/cluster"
+  filter_pattern  = module.aws_eks_audit_log.filter_pattern
+  destination_arn = module.aws_eks_audit_log.firehose_arn
+  depends_on      = [module.aws_eks_audit_log]
+}
+
+resource "aws_cloudwatch_log_subscription_filter" "lacework_cw_subscription_filter-us-west-2" {
+  for_each = toset(["cluster-3"])
+  provider = aws.us-west-2
+
+  name            = "${module.aws_eks_audit_log.filter_prefix}-${each.value}"
+  role_arn        = module.aws_eks_audit_log.cloudwatch_iam_role_arn
+  log_group_name  = "/aws/eks/${each.value}/cluster"
+  filter_pattern  = module.aws_eks_audit_log.filter_pattern
+  destination_arn = module.aws_eks_audit_log.firehose_arn
+  depends_on      = [module.aws_eks_audit_log]
+}
+```
```

Comment:
I know there are a bunch of annoyances with iteration in Terraform, but would it maybe be possible for us to have an input map that does something like this:

- Region
   - Cluster Name(s) array i.e. [cluster_name1, cluster_name2]
   - Provider Name

Then we could potentially iterate through each desired region, create the subscriptions for each cluster with the given provider.

Reply:
Unfortunately Terraform doesn't let us pass the provider in this way 😢